contracts/common/src/fixedi128.rs

```diff
@@ -42,6 +42,14 @@ impl FixedI128 {
             .map(FixedI128)
     }
 
+    pub fn to_precision(self, precision: u32) -> Option<i128> {
+        let prec_denom = 10i128.checked_pow(precision)?;
+
+        self.0
```

Comment:
I think we need to add some guard-flag that will prohibit any aritmetic operation after `to_precision`` call